src/services/sync/strategies/TSP/TSPBaseSyncer.js

```diff
@@ -128,17 +128,10 @@ class TSPBaseSyncer extends Syncer {
 		this.stats.schools.entities.push({ identifier, name });
 		try {
 			this.logInfo(`Finding school '${name}' (${identifier})...`);
-			const schools = await this.app.service('schools').find({
-				query: {
-					source: ENTITY_SOURCE,
-					'sourceOptions.schoolIdentifier': identifier,
-					$limit: 1,
-				},
-				paginate: false,
-			});
-			if (Array.isArray(schools) && schools.length === 1) {
+			const school = findSchool(this.app, identifier);
+			if (school) {
 				this.logInfo(`Updating '${name}' (${identifier})...`);
-				result = await this.updateSchool(schools[0], name);
+				result = await this.updateSchool(school, name);
```

Comment:
Should return result undefined in catch case? Is it is possible that it can run in catch case?

Reply:
I moved the error handling code on level up, so that this would be caught. Thanks!